src/main/java/com/duke/Duke.java

```diff
@@ -0,0 +1,197 @@
+                    String ddl = line.substring(line.indexOf("(")+1,line.indexOf(")"));
+                    list.add(new Event(description,ddl));
+                }else if (line.contains("[D]")) {
+                    String description = line.substring(6, line.indexOf("("));
+                    String ddl = line.substring(line.indexOf("(")+1,line.indexOf(")"));
+                    list.add(new Deadline(description,ddl));
+                }
+                list.get(list.size()-1).setDone(line.contains("[X]"));
+            }
+        } catch (FileNotFoundException e) {
+            dukePrint("unable to locate file");
+        }catch (IOException e) {
+            dukePrint("unable to create file");
+        }
+    }
+
+
+    public boolean getStatus() {
+        return active;
+    }
+
+
+    public void printLine() {
+        System.out.println(LINESEPARATER);
+    }
+
+
+    public void endLine() {
+        int max_len= 0 ;
+        for(int i = 0 ;i < dukeLineCount; i++) {
+            if(dukeLines[i].length() > max_len) {
+                max_len = dukeLines[i].length();
+            }
+        }
+        for(int i = 0 ;i < dukeLineCount; i++) {
+            for(int j=0; j<LINESEPARATER.length()-max_len+3; j++)
+            {
+                System.out.print(" ");
+            }
+            System.out.println(dukeLines[i]);
+        }
+        printLine();
+        System.out.print("\t");
+        dukeLineCount = 0;
+    }
+
+
+    public void endDuke() {
+
+        dukePrint("Bye. Hope to see you again soon!");
+        endLine();
+        active = false;
+    }
+
+
+    public void greet() {
+        dukePrint("Hello! I'm Duke");
+        dukePrint("What can I do for you?");
+        endLine();
+    }
+
+
+    public void unknownAction() {
+        dukePrint("Sorry! I don't understand");
+        endLine();
+    }
+
+
+    public void addList(Task item) {
+        list.add(item);
+        dukePrint("Got it. I've added this task:");
+        dukePrint(list.get(list.size()-1).toString());
+        dukePrint("Now you have " + list.size() + " tasks in the list.");
+        try {
+            //fileTask.addToFile(list.get(list.size()-1));
+            fileTask.rewriteFile(list);
+        }catch (IOException e) {
+            dukePrint("unable to write to file");
+        }
+        endLine();
+    }
+
+
+    public void listOut() {
+        if(list.size()==0) {
+            dukePrint("Woohooo no tasks due ~~~~");
+        }
+        else{
+            dukePrint("Now you have " + list.size() + " tasks in the list.");
+        }
+        for(int i = 0; i< list.size(); i++) {
+            dukePrint((i+1) + ". "+list.get(i).toString());
+        }
+        endLine();
+    }
+
+
+    public void markDone(String line) {
+        try {
+            line = line.replaceAll("[^(\\d)]", "");
+            int index = Integer.parseInt(line);
+            index -= 1;
+            list.get(index).setDone(true);
```

Comment:
This line is [doing two things](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#do-not-trip-up-reader), and same for line 155. How about naming both actions on separate lines?